### PR TITLE
Fixed Typo in NLLLoss Function docstring

### PR DESCRIPTION
**Pull request by helgehr**
The documentation of the function [`NLLLoss`](https://pytorch.org/docs/stable/generated/torch.nn.NLLLoss.html) currently contains that the loss for a sample n is `l_n = - w_{y_n} x_{n,y_n}`, this indicates that x has two dimensions which is not the case.
The correct form should be `l_n = - w_{y_n} x_n y_n`.

### CODE REVIEW

File: torch/nn/modules/loss.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ class NLLLoss(_WeightedLoss):
 
     .. math::
         \ell(x, y) = L = \{l_1,\dots,l_N\}^\top, \quad
-        l_n = - w_{y_n} x_{n,y_n}, \quad
+        l_n = - w_{y_n} x_n y_n, \quad
```
Comment (albanD) [inline torch/nn/modules/loss.py:126]: The `y`s (target) are indices in this class. So the current doc is correct no?

Comment (helgehr) [reply torch/nn/modules/loss.py:126]: `y_n` as an index for the weights seems reasonable but why should the input `x` have `y_n` as an index additional to `n`?

Comment (albanD) [reply torch/nn/modules/loss.py:126]: `n` is the index of the sample and `y_n` is reading the value of X at the ground truth class.
You can check the doc, the size of `x` is `(N, C)` (with N the batch size and C the number of classes).

Comment (helgehr) [reply torch/nn/modules/loss.py:126]: Ok thank you for the clarification, I misunderstood that - sorry.

Comment (albanD) [reply torch/nn/modules/loss.py:126]: No worries!

**Comment by albanD** [comment]
Closing as the current doc is accurate.
